### ISSUE TITLE
Poison smoke for cs:go

### ISSUE THREAD

**Issue by Hetachi** [peace-maker/smrpg#301]
Any chance to make a round area of affect where smoke landed in cs:go, and then apply damage to anyone within that area ?

**Comment by peace-maker** [comment]
Yes, sure. I've never added it, because I didn't want to hardcode the time the smoke takes to disappear so it stays aligned after game updates. CS:S has a nice entity for the smoke `env_particlesmokegrenade` with properties holding time when the smoke starts to fade. Last time I checked, CSGO didn't expose that information.

**Comment by baferpro** [comment]
`public Action Timer_SmokeDamage(Handle timer, any dp)
{
	static float distance = 200.0;
	
	ResetPack(dp);
	int client = CID(ReadPackCell(dp));
	if ( client && !g_bLR ) { 
		int entity = EntRefToEntIndex(ReadPackCell(dp));
		if ( entity > MaxClients ) {
			int times = ReadPackCell(dp);
			if ( times > 0 ) {
				float smokeOrigin[3], clientOrigin[3];
				GetEntPropVector(entity, Prop_Send, "m_vecOrigin", smokeOrigin);
				float damage = smoke_multipliers[GetSkillLevel(client, "smoke")];
				
				LC(i) { 
					if ( IsPlayerAlive(i) && i != client ) {
						GetClientAbsOrigin(i, clientOrigin);
						if ( GetVectorDistance(smokeOrigin, clientOrigin) <= distance ) {
							SDKHooks_TakeDamage(i, client, client, damage);
						}
					}
				}
				
				return Plugin_Continue;
			}
		}
	}
	
	CloseHandle(dp);
	return Plugin_Stop;
}`
one of the working parts of the code in the CS:GO
